pkg/tfbridge/internal/schemaconvert/schemaconvert.go

```diff
@@ -0,0 +1,134 @@
+package schemaconvert
+
+import (
+	v1Schema "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	v2Schema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+func Sdkv2ToV1Type(t v2Schema.ValueType) v1Schema.ValueType {
+	return v1Schema.ValueType(t)
+}
+
+func Sdkv2ToV1SchemaOrResource(elem interface{}) interface{} {
+	switch elem := elem.(type) {
+	case nil:
+		return nil
+	case *v2Schema.Schema:
+		return Sdkv2ToV1Schema(elem)
+	case *v2Schema.Resource:
+		return Sdkv2ToV1Resource(elem)
+	default:
+		contract.Failf("unexpected type %T", elem)
+		return nil
+	}
+}
+
+func Sdkv2ToV1Resource(sch *v2Schema.Resource) *v1Schema.Resource {
+	//nolint:staticcheck
+	if sch.MigrateState != nil {
+		contract.Failf("MigrateState is not supported in conversion")
+	}
```

Comment:
This is an assertion.
```suggestion
	contract.Assertf(sch.MigrateState == nil, "MigrateState is not supported in conversion")
```

Let's do the same for all of the other `if !T { contract.Failf(...) }`.